App/Containers/StatusCheck.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react'
+import { ActivityIndicator, View } from 'react-native'
+import { connect } from 'react-redux'
+import { NavigationScreenProps } from 'react-navigation'
+
+import FatalErrorView from '../Components/FatalErrorView'
+
+import { RootState } from '../Redux/Types'
+import { NodeState } from '../Redux/TextileNodeRedux'
+
+const MIGRATION_NEEDED_ERROR = 'repo needs migration'
+
+interface StateProps {
+  onboarded: boolean
+  nodeStarted: boolean
+  nodeError?: string
+}
+
+type Props = StateProps & NavigationScreenProps<{}>
+
+class StatusCheck extends React.Component<Props, {}> {
```

Comment:
This is basically what `OnboardingCheck` was, just using it as a general use hub for checking state and making decisions about where to navigate. This adds the option of loading the migration flow.